core/Tracker/GoalManager.php

```diff
@@ -742,6 +742,11 @@ protected function insertNewConversion($conversion, $visitInformation, Request $
         Common::printDebug($newGoalDebug);
 
         $wasInserted = $this->getModel()->createConversion($conversion);
+        if (!$wasInserted) {
+            $idSite = $request->getIdSite();
+            $idorder = $request->getParam('ec_id');
+            throw new \Exception("Invalid non-unique idsite/idorder combination ($idSite, $idorder), conversion was not inserted.");
```

Comment:
@diosmosis here we cannot be 100% sure it failed due to the idOrder? Maybe we trigger the exception only when `ec_id` was set? 

Not sure if we want to trigger an exception in the other cases as it may prevent the tracking request to be fully processed as the `recordLogs` of other processors may not be executed anymore? Eg when it is a pageview and then triggers a conversion... on the other side we would never notice the conversion isn't being tracked. Tricky...

Also I think it should be maybe a `Piwik\Exception\InvalidRequestParameterException`? This should make sure to send the correct status code here: https://github.com/matomo-org/matomo/blob/3.9.0-b3/core/Tracker/Handler.php#L89-L91

Reply:
Makes sense, will update.